sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalogSuite.scala

```diff
@@ -1119,4 +1119,32 @@ class V2SessionCatalogNamespaceSuite extends V2SessionCatalogBaseSuite {
     }
     catalog.dropNamespace(testNs, cascade = false)
   }
+
+  test("IdentifierHelper should throw exception when identifier has more than 2 parts") {
```

Comment:
Don't this test and below duplicate the test in QueryCompilationErrorsSuite.scala?

Reply:
`QueryCompilationErrorsSuite` - tests `MultipartIdentifierHelper.asTableIdentifier`,  
`JDBCV2Suite` - tests `IdentifierHelper.asFunctionIdentifier`
You are right there are some overlapping...
I've removed testing of these 2 functions from this file, left testing only of `IdentifierHelper.asTableIdentifier` and `MultipartIdentifierHelper.asFunctionIdentifier` functions, which is not possible to reproduce from user side (sql)...